### PR TITLE
Domains: Add pending transfer notification

### PR DESCRIPTION
**Pull request by aidvu**
- Add pending transfer domain warning
- Add pending_transfer field to assembler

Test live: https://calypso.live/?branch=add/pending-transfer-notification

### CODE REVIEW

File: client/my-sites/upgrades/components/domain-warnings/index.jsx [hunk 1]
```diff
@@ -345,6 +347,39 @@ export default React.createClass( {
...
+			fullNotice = this.translate( '%(domain)s is pending transfer. You must wait for the transfer to finish, and then update the settings at the new registrar.', {
+				args: {
+					domain: domain.name
+				}
+			} );
+
+		return (
+			<Notice
+				isCompact={ this.props.isCompact }
+				status="is-error"
```
Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: Isn't `error` a bit too much? Warning at most, maybe even just info or default should suffice.

File: client/my-sites/upgrades/components/domain-warnings/index.jsx [hunk 2]
```diff
@@ -345,6 +347,39 @@ export default React.createClass( {
 				section="domain-management" />;
 	},
 
+	pendingTransfer() {
+		const domains = this.getDomains().filter( domain => domain.pendingTransfer );
+
+		if ( domains.length !== 1 ) {
```
Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: While probably not the most common case, but what if the user has more than one pending transfer domains? I can imagine a user that has two custom domains decides to move from wpcom somewhere else and initiates transfers of both domains. And, boom, no notice ;)

File: client/my-sites/upgrades/components/domain-warnings/index.jsx [hunk 3]
```diff
@@ -345,6 +347,39 @@ export default React.createClass( {
...
+		}
+
+		const domain = domains[ 0 ];
+
+		const compactNotice = this.translate( '%(domain)s is pending transfer.', {
+				args: {
+					domain: domain.name
+				}
+			} ),
+			fullNotice = this.translate( '%(domain)s is pending transfer. You must wait for the transfer to finish, and then update the settings at the new registrar.', {
```
Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: This line is way too long (don't make a "guess what's also too long" joke, don't make a "guess what's also too long" joke... I've been watching How I Met Your Mother too much lately, especially Barney Stinson).

File: client/my-sites/upgrades/components/domain-warnings/index.jsx (hunk 2 above)
Comment (aidvu) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: Truth is, I wanted this to be visible only when a users enters `edit` view. :) @klimeryk

**Comment by klimeryk** [comment]
screen shot 2016-08-29 at 22 07 54
You can still navigate to the `Contacts and Privacy` view by clicking the button on the Privacy Protection row in the domain's details view.

And my usual mantra - please keep the domain bold and beautiful ;)

File: client/my-sites/upgrades/components/domain-warnings/index.jsx (hunk 2 above)
Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: Ah, I see - makes sense 👍 Then my only concern is that someone could add this rule to the domains list view at some point in time and then it won't behave as one might think. Maybe a comment warning about that would suffice here (though, who reads comments? 😝 ).

File: client/my-sites/upgrades/components/domain-warnings/index.jsx [hunk 4]
```diff
@@ -519,6 +520,40 @@ export default React.createClass( {
...
+			return null;
+		}
+
+		const domain = domains[ 0 ];
+
+		const compactNotice = this.translate( '{{strong}}%(domain)s{{/strong}} is pending transfer.', {
+				components: { strong: <strong /> },
+				args: { domain: domain.name }
+			} ),
+			fullNotice = this.translate(
```
Comment (aidvu) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: Don't think a comment is needed. This rule is not in the default whitelist, so it has to be passed through props.

File: client/my-sites/upgrades/components/domain-warnings/index.jsx (hunk 2 above)
Comment (aidvu) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: Don't think a comment is needed. This rule is not in the default whitelist, so it has to be passed through props.
 Add a line note

**Comment by aidvu** [comment]
@klimeryk ready for a new round. :)

File: client/my-sites/upgrades/components/domain-warnings/index.jsx [hunk 5]
```diff
@@ -519,6 +520,40 @@ export default React.createClass( {
...
+				'{{strong}}%(domain)s{{/strong}} is pending transfer. ' +
+				'You must wait for the transfer to finish, and then update the settings at the new registrar.',
+				{
+					components: { strong: <strong /> },
```
Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: ❤️

**Comment by klimeryk** [comment]
Nice! 🚢 

Just a side note - I love that you've fixed quite a few small issues (especially making those `renewLink`s const) while you were at it, but please keep such refactors in a separate commit. It makes it so much easier to review and to just see what part of the PR is a refactor and what's the actual change :) Same goes for #7727 :)

**Comment by aidvu** [comment]
Totally agreed. Thanks! :)

**Pull request merged**